tests/user-envs-migration.sh

```diff
@@ -25,11 +25,13 @@ nix-env -f user-envs.nix -i bar-0.1
 # Migrate to the new profile dir, and ensure that everything’s there
 export PATH="$PATH_WITH_NEW_NIX"
 nix-env -q # Trigger the migration
-( [[ -L ~/.nix-profile ]] && \
-    [[ $(readlink ~/.nix-profile) == ~/.local/share/nix/profiles/profile ]] ) || \
-    fail "The nix profile should point to the new location"
-
-(nix-env -q | grep foo && nix-env -q | grep bar && \
-    [[ -e ~/.nix-profile/bin/foo ]] && \
-    [[ $(nix-env --list-generations | wc -l) == 2 ]]) ||
-    fail "The nix profile should have the same content as before the migration"
+
+# The nix profile should point to the new location
+[[ -L ~/.nix-profile ]]
+[[ $(readlink ~/.nix-profile) == ~/.local/state/nix/profiles/profile ]]
```

Comment:
Wait, that section is actually completely broken, I removed this migration from the original PR after some review suggested it :thinking: . So `~/.nix-profile` should still point to the original location

Reply:
Ha! Sorry, I should have marked this as draft as I opened it before tests. So I guess we can instead test that the migration doesn't happen until later OK.

Will rework.

(And I realized how I can run the tests incrementally, just define the env var. Whew. That helps.)